### PR TITLE
Ticket 98 type script less any

### LINKED ISSUE #98: Remove more `any`s in typescript code
Remove as much `any` in typescript code of IMA Agent as possible

### PR DESCRIPTION
**Pull request by sergiy-skalelabs**
Massively removing `any`-kwyword usage from **TypeScript** code.
No new features introduced.
No performance changes expected.
No new tests needed.

### CODE REVIEW

File: src/imaEventLogScan.ts [hunk 1]
```diff
@@ -228,15 +248,15 @@ export async function safeGetTransactionCount(
...
             return ret;
         } catch ( err ) {
-            ret = retValOnFail;
+            ret = retValOnFail ?? owaspUtils.toBN( 0 );
             details.error( "Failed call attempt {} of {} to {} via {url}, error is: {err}, " +
                 "stack is:\n{stack}", idxAttempt, cntAttempts, strFnName + "()", u, err, err );
         }
     }
-    if( ( idxAttempt + 1 ) > cntAttempts && ret === "" ) {
+    if( ( idxAttempt + 1 ) > cntAttempts && ( !ret ) ) {
```
Comment (DmytroNazarenko) [inline src/imaEventLogScan.ts]: Remove redundant `()`

File: src/loopWorker.ts [hunk 2]
```diff
@@ -73,7 +76,8 @@ class ObserverServer extends SocketServer {
         self.intervalPeriodicSchainsCaching = null;
         self.bIsPeriodicCachingStepInProgress = false;
         self.mapApiHandlers.init =
-        function( joMessage: any, joAnswer: any, eventData: any, socket: any ): any {
+        function( joMessage: state.TLoadedJSON, joAnswer: any,
```
Comment (DmytroNazarenko) [inline src/loopWorker.ts:79]: use object instead of any

File: src/oracle.ts [hunk 3]
```diff
@@ -162,10 +179,11 @@ async function handleOracleSubmitRequestResult(
 }
 
 export async function oracleGetGasPrice(
-    oracleOpts: any, details: log.TLogger ): Promise<any> {
+    oracleOpts: TOracleOptions, details: log.TLogger
```
Comment (DmytroNazarenko) [inline src/oracle.ts]: Use log.TLoggerBase instead

File: src/state.ts [hunk 4]
```diff
@@ -527,6 +613,6 @@ export function isPreventExitAfterLastAction(): boolean {
     return gFlagIsPreventExitAfterLastAction;
 }
 
-export function setPreventExitAfterLastAction( isPrevent: any ): void {
+export function setPreventExitAfterLastAction( isPrevent?: boolean | null ): void {
```
Comment (DmytroNazarenko) [inline src/state.ts]: Redundant null

File: src/eventDispatcher.ts [hunk 5]
```diff
@@ -23,9 +23,13 @@
  * @copyright SKALE Labs 2019-Present
  */
 
+export type TType = any;
+export type TListener = any;
+export type TData = any;
```
Comment (DmytroNazarenko) [inline src/eventDispatcher.ts]: Use object instead of any

File: src/imaTx.ts [hunk 6]
```diff
@@ -40,44 +40,53 @@ import * as imaEventLogScan from "./imaEventLogScan.js";
...
+    gasPrice?: string
+    gasLimit?: string
+    value?: string
+};
+
 export interface TCustomPayedCallOptions {
     isCheckTransactionToSchain?: boolean
 }
 
+export declare type TPayedCallMethodArgument = any;
```
Comment (DmytroNazarenko) [inline src/imaTx.ts:54]: Try to replace any

**Review by DmytroNazarenko — APPROVED** [review]

**Review by DmytroNazarenko — CHANGES_REQUESTED** [review]
Need to fix issues above

File: src/imaEventLogScan.ts (hunk 1 above)
Comment (sergiy-skalelabs) [reply src/imaEventLogScan.ts]: fixed

File: src/oracle.ts (hunk 3 above)
Comment (sergiy-skalelabs) [reply src/oracle.ts]: fixed

File: src/state.ts (hunk 4 above)
Comment (sergiy-skalelabs) [reply src/state.ts]: fixed

File: src/eventDispatcher.ts (hunk 5 above)
Comment (sergiy-skalelabs) [reply src/eventDispatcher.ts]: fixed

File: src/imaTx.ts (hunk 6 above)
Comment (sergiy-skalelabs) [reply src/imaTx.ts:54]: Here it's honest `any`. It's any argument of any method on any Ethereum-like network

File: src/loopWorker.ts (hunk 2 above)
Comment (sergiy-skalelabs) [reply src/loopWorker.ts:79]: Here it's more-less honest `any`

**Review by DmytroNazarenko — APPROVED** [review]

**Pull request merged**